[stdlib][NFC] Put stored properties & primary initializers before other members in all struct declarations (#39608)
swiftlang/swift: docs/StandardLibraryProgrammersManual.md

### Formatting Conventions

The Standard Library codebase has some rather strict formatting conventions. While these aren't currently automatically enforced, we still expect these conventions to be followed in every PR, including draft PRs. (PRs are first and foremost intended to be read/reviewed by *people familiar with the stdlib codebase*, and it's crucial that trivial formatting issues don't get in the way of understanding proposed changes.)

#### Line Breaking

The stdlib currently has a hard line length limit of 80 characters. This allows code to be easily read in environments that don't gracefully handle long lines, including (especially!) code reviews on GitHub.
   
We use two spaces as the unit of indentation. We don't use tabs.

To break long lines, please closely follow the indentation conventions you see in the existing codebase. (FIXME: Describe in detail.)

> Our primary rule is that if we need to break a list (such as arguments, tuple or array/dictionary literals, generic type parameters, etc.), then we always put each item on its own line, indented by +1 unit, even if a series of items would fit on a single line together. 

Author: Oh that's a good point. I will adjust the wording.

> Our primary rule is that if we need to break a list (such as arguments, tuple or array/dictionary literals, generic type parameters, etc.), then we always put each item on its own line, indented by +1 unit, even if a series of items would fit on a single line together. 

Author: (I also added an explicit example for breaking a list only at the delimiters.)

- In some cases, it can also work well to declare the most essential protocol conformances directly on the type definition; feel free to do so if it helps understanding. (You can still implement requirements in separate extensions in this case, or you can do it within the main declaration.)

- It's okay for the core type declaration to forward reference large nested types or static members that are defined in subsequent extensions. It's often a good idea to define these in an extension immediately following the type declaration, but this is not a strict rule. 

Extensions are a nice way to break up the implementation into easily digestable chunks, but they aren't the only way. The goal is to make things easy to understand -- if a type is small enough, it may be best to list every member directly in the `struct`/`class` definition, while for huge types it often makes more sense to break them up into number of separate source files. 

> ```

Author: ```suggestion
```swift
```

To break long lines, please closely follow the indentation conventions you see in the existing codebase. (FIXME: Describe in detail.)

> Our primary rule is that if we need to put a line break anywhere in the middle of a list (such as arguments, tuple or array/dictionary literals, generic type parameters, etc.), then we need to go the full way, and put each item on its own line, indented by +1 unit, even if some of the items would fit on a single line together.

Reviewer: Nit:
```suggestion
Our primary rule is that if we need to put a line break anywhere in the middle of a list (such as arguments, tuple or array/dictionary literals, generic type parameters, etc.), then we should go all the way, and put each item on its own line, indented by +1 unit, even if some of the items would fit on a single line together.
```

Reviewer: Another one:
```suggestion
Our primary rule is that if we need to insert a line break anywhere in the middle of a list (such as arguments, tuple or array/dictionary literals, generic type parameters, etc.), then we need to go the full way, and put each item on its own line, indented by +1 unit, even if some of the items would fit on a single line together.
```

Author: I sneakily edited your first suggestion to change "then we need to go" to "then we must go" (RFC 2119 style), to emphasize that this is not something people can opt out of if they don't like it. 🤓
